app/controllers/works_controller.rb

```diff
@@ -37,50 +40,78 @@ def show
     @votes = @work.votes.order(created_at: :desc)
   end
 
+  #only the user who created the work can edit
   def edit
+    unless @work.can_edit_or_delete_work?(session[:user_id])
+      redirect_to work_path(@work.id)
```

Comment:
It would be good to add a filter or at least an if statement to make sure a user can't access this action unless they are the user who created the work.